test/Common.FunctionalTests/AppOfflineTests.cs

```diff
@@ -67,6 +150,10 @@ public class AppOfflineTests : IISFunctionalTestBase
                 RemoveAppOffline(deploymentResult.DeploymentResult.ContentRoot);
                 await AssertRunning(deploymentResult);
             }
+
+            AddAppOffline(deploymentResult.DeploymentResult.ContentRoot);
+            await AssertAppOffline(deploymentResult);
+            AssertFilesNotLocked(deploymentResult);
```

Comment:
Since `AddAppOffline` and `AssertAppOffline` are already run 5 times in the loop above, I assume this new code is really about `AssertFilesNotLocked`.  If so, why not just add this inside the existing loop, rather than adding another iteration with this one additional check?

Reply:
Because `AssertFilesNotLocked` deletes published output so there would be no restarting the site.

Reply:
OK, I recommend renaming this method to make it clear it deletes the output.  `Assert*` methods are typically assumed to have no side effects.